### ISSUE TITLE
[Screen reader] Incorrect aria-label after entering an invalid value in input

### ISSUE THREAD

**Issue by microbit-robert** [RaspberryPiFoundation/blockly#9466]
### Check for duplicates

- [x] I have searched for similar issues before opening a new one.

### Description

Something goes wrong when editing a number field if you enter an invalid value. If you edit a number field by entering "3whoops", the input is invalid. The input value reverts to what it previously was (e.g., 400), however, the aria-label for the input now includes "3" and not "400". This doesn't seem to go wrong if the number is at the end of the invalid input. Refocusing the input doesn't correct the aria-label. I don't know how far this extends to other inputs or what other values the user can enter to make this fail. 

### Reproduction steps

1. Go to the [screen reader test page](https://raspberrypifoundation.github.io/blockly-keyboard-experimentation/screenreader/)
2. Turn on a screen reader and navigate to or inspect the aria-label for the "width" input in the "create canvas" block and note that it is currently "400"
3. Edit the field and enter "3whoops" - this is invalid and when hitting "Enter" the value will revert to "400"
4. Note that the aria-label for the input is now "3" instead of "400"

### Stack trace

```shell

```

### Screenshots

https://github.com/user-attachments/assets/8b849f59-c3f7-4c6e-ab5e-524b22004aa9

**Comment by BenHenning** [comment]
Wow. I'm pretty sure I tested this flow a while back but we've made a bunch of changes to field labeling and it probably broke (or maybe I didn't check going back to the field). Nice catch!

**Comment by BenHenning** [comment]
Sent #9469 to fix. Fortunately a very easy fix.

**Comment by BenHenning** [comment]
Fixed in #9469.
